### ISSUE TITLE
A remind for multi-channel feature maps (such as rgb pictures)

### ISSUE THREAD

**Issue by LosNowveShanghaiTech** [Po-Hsun-Su/pytorch-ssim#33]
If you want to use this loss for multi-channel feature maps (such as RGB pictures), please calculate the loss of each channel separately and sum them up. (due to F.conv2d)

**Comment by tangbao** [comment]
Thank you, this helps.

Does it work for a batch of images?

**Comment by LosNowveShanghaiTech** [comment]
> Thank you, this helps.
> 
> Does it work for a batch of images?

Sorry, due to my carelessness, I misunderstood the cause of a bug. During my debugging process yesterday, I found that the real cause is that under the accuracy of FLOAT32, the calculation of variance will have a negative value, and the two hyperparameters C1 and C2 are irreparable. I tried to modify L (maximum brightness) from 255 to 1; kept C1 and C2 unchanged, the problem was solved.
The author's code not only supports multi-channel pictures but also supports "multi-batch and multi-channel" pictures. For example, pictures with size (8, 3, 256, 256) are acceptable.

**Comment by LosNowveShanghaiTech** [comment]
Another reminder: If you want to use SSIM loss (not multi-scale SSIM) to optimize a set of pictures, you need to pay attention to the part that has no structure (such as a solid color area larger than window_size), although a loss will occur, the **derivative is 0**, so this part cannot be optimized.
